### ISSUE TITLE
luarocks install lua rock exception

### ISSUE THREAD

**Issue by LiuYawei2019** [luarocks/luarocks#1489]
### Please read the following before submitting:
- Please do NOT submit bug reports for questions. Ask questions in the [Gitter
  chat-room](https://gitter.im/luarocks/luarocks).
- Please do NOT submit duplicate bug reports. Look through the [issue
  tracker](https://github.com/luarocks/luarocks/issues) before submitting.

### Please fill out the following:
- **Platform:**
  - MacOS.

- **LuaRocks version:**
  - `luarocks --version`
3.9.1
- **Configuration file:**
 ```
accept_unknown_fields = false
arch = "macosx-x86_64"
cache = {
   luajit_version_checked = true
}
cache_fail_timeout = 86400
cache_timeout = 60
check_certificates = false
cmake_generator = "Unix Makefiles"
config_files = {
   nearest = "/Users/hero/.luarocks/config-5.4.lua",
   system = {
      file = "/usr/local/etc/luarocks/config-5.4.lua",
      found = true
   },
   user = {
      file = "/Users/hero/.luarocks/config-5.4.lua",
      found = true
   }
}
connection_timeout = 30
deploy_bin_dir = "/usr/local/bin"
deploy_lib_dir = "/usr/local/lib/lua/5.4"
deploy_lua_dir = "/usr/local/share/lua/5.4"
deps_mode = "one"
disabled_servers = {}
export_path_separator = ":"
external_deps_dirs = {
   "/usr/local",
   "/usr",
   "/",
   "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr",
   "/usr/local/opt"
}
external_deps_patterns = {
   bin = {
      "?"
   },
   include = {
      "?.h"
   },
   lib = {
      "/?/lib/lib?.dylib",
      "lib?.tbd",
      "lib?.a",
      "lib?.so",
      "lib?.so.*"
   }
}
external_deps_subdirs = {
   bin = "bin",
   include = "include",
   lib = {
      "",
      "lib"
   }
}
external_lib_extension = "dylib"
fs_use_modules = true
gcc_rpath = false
home = "/Users/hero"
home_tree = "/Users/hero/.luarocks"
homeconfdir = "/Users/hero/.luarocks"
hooks_enabled = true
lib_extension = "so"
lib_modules_path = "/lib/lua/5.4"
link_lua_explicitly = false
local_by_default = false
local_cache = "/Users/hero/.cache/luarocks"
lua_extension = "lua"
lua_found = {
   lua_bindir = "/usr/local/opt/lua/bin",
   lua_dir = "/usr/local/opt/lua",
   lua_interpreter = "lua5.4",
   lua_version = "5.4"
}
lua_interpreter = "lua5.4"
lua_modules_path = "/share/lua/5.4"
lua_version = "5.4"
no_manifest = false
obj_extension = "o"
processor = "x86_64"
program_version = "3.9.1"
rocks_dir = "/usr/local/lib/luarocks/rocks-5.4"
rocks_servers = {
   {
      "https://luarocks.org",
      "https://raw.githubusercontent.com/rocks-moonscript-org/moonrocks-mirror/master/",
      "https://luafr.org/luarocks/"
   }
}
rocks_subdir = "/lib/luarocks/rocks-5.4"
rocks_trees = {
   {
      name = "user",
      root = "/Users/hero/.luarocks"
   },
   {
      name = "system",
      root = "/usr/local"
   }
}
runtime_external_deps_patterns = {
   bin = {
      "?"
   },
   include = {
      "?.h"
   },
   lib = {
      "/?/lib/lib?.dylib",
      "lib?.tbd",
      "lib?.so",
      "lib?.so.*"
   }
}
runtime_external_deps_subdirs = {
   bin = "bin",
   include = "include",
   lib = {
      "",
      "lib"
   }
}
static_lib_extension = "a"
sysconfdir = "/usr/local/etc/luarocks"
target_cpu = "x86_64"
upload = {
   api_version = "1",
   server = "https://luarocks.org",
   tool_version = "1.0.0"
}
user_agent = "LuaRocks/3.9.1 macosx-x86_64"
variables = {
   AR = "ar",
   BUNZIP2 = "bunzip2",
   CC = "env MACOSX_DEPLOYMENT_TARGET=11.0 gcc",
   CFLAGS = "-O2 -fPIC",
   CHMOD = "chmod",
   CMAKE = "cmake",
   CP = "cp",
   CURL = "curl",
   CURLNOCERTFLAG = "-k",
   CVS = "cvs",
   FIND = "find",
   GIT = "git",
   GPG = "gpg",
   GUNZIP = "gunzip",
   HG = "hg",
   ICACLS = "icacls",
   LD = "env MACOSX_DEPLOYMENT_TARGET=11.0 gcc",
   LIBFLAG = "-bundle -undefined dynamic_lookup -all_load",
   LIB_EXTENSION = "so",
   LS = "ls",
   LUA = "/usr/local/opt/lua/bin/lua5.4",
   LUALIB = "liblua5.4.dylib",
   LUA_BINDIR = "/usr/local/opt/lua/bin",
   LUA_DIR = "/usr/local/opt/lua",
   LUA_INCDIR = "/usr/local/opt/lua/include/lua5.4",
   LUA_LIBDIR = "/usr/local/opt/lua/lib",
   LUA_LIBDIR_FILE = "liblua5.4.dylib",
   MAKE = "make",
   MD5 = "md5",
   MD5SUM = "md5sum",
   MKDIR = "mkdir",
   MKTEMP = "mktemp",
   OBJ_EXTENSION = "o",
   OPENSSL = "openssl",
   OPENSSL_INCDIR = "/usr/local/opt/openresty-openssl111/include",
   OPENSSL_LIBDIR = "/usr/local/opt/openresty-openssl111/lib",
   PWD = "pwd",
   RANLIB = "ranlib",
   RM = "rm",
   RMDIR = "rmdir",
   ROCKS_TREE = "/usr/local/lib/luarocks/rocks-5.4",
   RSYNC = "rsync",
   RSYNCFLAGS = "--exclude=.git -Oavz",
   SCP = "scp",
   SCRIPTS_DIR = "/usr/local/bin",
   SEVENZ = "7z",
   SSCM = "sscm",
   SVN = "svn",
   TAR = "tar",
   TEST = "test",
   TOUCH = "touch",
   UNZIP = "unzip -n",
   WGET = "wget",
   WGETNOCERTFLAG = "--no-check-certificate",
   ZIP = "zip"
}
web_browser = "open"
wrapper_suffix = ""
```
- **LuaRocks output from when the issue occurred:**
  - Use the `--verbose` flag.

- **Description:**
  - The steps you took in plain English to reproduce the problem.
  - use homebrew install luarocks，default luarocks dependencis lua  version 5.4
  - when execute luarocks install dkjson , exception infomations :
```
Error: LuaRocks 3.9.1 bug (please report at https://github.com/luarocks/luarocks/issues).
Arch.: macosx-x86_64
/usr/local/share/lua/5.4/ssl/https.lua:66: bad argument #2 to 'method' (string expected, got light userdata)
stack traceback:
        [C]: in function 'socket.http.request'
        /usr/local/share/lua/5.4/ssl/https.lua:132: in function 'ssl.https.request'
        .../Cellar/luarocks/3.9.1/share/lua/5.4/luarocks/fs/lua.lua:739: in upvalue 'request'
        .../Cellar/luarocks/3.9.1/share/lua/5.4/luarocks/fs/lua.lua:847: in upvalue 'http_request'
        .../Cellar/luarocks/3.9.1/share/lua/5.4/luarocks/fs/lua.lua:914: in function 'luarocks.fs.lua.download'
        (...tail calls...)
        ...l/Cellar/luarocks/3.9.1/share/lua/5.4/luarocks/fetch.lua:154: in function 'luarocks.fetch.fetch_url'
        ...l/Cellar/luarocks/3.9.1/share/lua/5.4/luarocks/fetch.lua:50: in function 'luarocks.fetch.fetch_caching'
        ...l/Cellar/luarocks/3.9.1/share/lua/5.4/luarocks/manif.lua:108: in function 'luarocks.manif.load_manifest'
        .../Cellar/luarocks/3.9.1/share/lua/5.4/luarocks/search.lua:111: in function <.../Cellar/luarocks/3.9.1/share/lua/5.4/luarocks/search.lua:101>
        (...tail calls...)
        .../Cellar/luarocks/3.9.1/share/lua/5.4/luarocks/search.lua:155: in function 'luarocks.search.search_repos'
        .../Cellar/luarocks/3.9.1/share/lua/5.4/luarocks/search.lua:241: in function 'luarocks.search.find_suitable_rock'
        ...al/Cellar/luarocks/3.9.1/share/lua/5.4/luarocks/deps.lua:226: in function 'luarocks.deps.fulfill_dependency'
        ...al/Cellar/luarocks/3.9.1/share/lua/5.4/luarocks/deps.lua:336: in function 'luarocks.deps.fulfill_dependencies'
        ...l/Cellar/luarocks/3.9.1/share/lua/5.4/luarocks/build.lua:138: in upvalue 'process_dependencies'
        ...l/Cellar/luarocks/3.9.1/share/lua/5.4/luarocks/build.lua:379: in function 'luarocks.build.build_rockspec'
        (...tail calls...)
        ...llar/luarocks/3.9.1/share/lua/5.4/luarocks/cmd/build.lua:154: in function 'luarocks.cmd.build.command'
        (...tail calls...)
        [C]: in function 'xpcall'
        ...cal/Cellar/luarocks/3.9.1/share/lua/5.4/luarocks/cmd.lua:672: in function 'luarocks.cmd.run_command'
        /usr/local/bin/luarocks:38: in main chunk
        [C]: in ?
        make: *** [deps] Error 99
```

**Comment by hishamhm** [comment]
This is a duplicate of #1422. See more info here: https://github.com/luarocks/luarocks/issues/1422#issuecomment-1172649206
